src/core/thread/mle_router.cpp

```diff
@@ -1309,7 +1310,7 @@ ThreadError MleRouter::HandleAdvertisement(const Message &aMessage, const Ip6::M
     case kDeviceStateChild:
         if ((mDeviceMode & ModeTlv::kModeFFD) && (GetActiveRouterCount() < mRouterUpgradeThreshold))
         {
-            BecomeRouter(ThreadStatusTlv::kTooFewRouters);
+            mRouterSelectionJitterTimer.Start(Timer::SecToMsec(otPlatRandomGet() % kRouterSelectionJitter));
```

Comment:
Certificate tests dose not require and check this jitter during attaching explicitly(at least for current Harness version R19), but I notice that for reference unit or DUT as Router, Harness always wait for REED to upgrade to Router for 125s prior to execution. Please let me file a JIRA to clarify this point first.